.travis.yml

```diff
@@ -49,7 +49,7 @@ install:
 # Nek5k
   - git clone --depth 1 https://github.com/Nek5000/Nek5000.git;
   - cd Nek5000/tools && ./maketools genbox genmap reatore2 && cd ../..;
-  - export NEK5K_DIR=$PWD/Nek5000 PATH=$PWD/Nek5000/bin:$PATH;
+  - export NEK5K_DIR=$PWD/Nek5000 NEK_BOX_DIR=./build/boxes PATH=$PWD/Nek5000/bin:$PATH;
```

Comment:
I don't like this because `make prove-all` doesn't work unless the user manually sets `NEK_BOX_DIR`. Does this need to be a default set in `Makefile`?

Reply:
I think this issue is why I previously set the scrit to build the boxes in the `examples/nek5000` dir and copied them over rather than trying to build them in the `build` dir. It avoids this special case.

Reply:
Surely it's possible for `genbox` to put the boxes in a location provided as a command-line argument?

Reply:
I'm not sure, but we can definitely make it so for this script. @thilinarmtb, I also noticed that `make examples` fails, though it does it on master as well.

Current status of ways to build the Nek examples in this branch:

`make examples` from top level - fails
```
make: *** No rule to make target 'build/nek-bp1', needed by 'examples'.  Stop.
```

`make nek` in `examples` folder - fails 
```cd nek5000 && NEK5K_DIR=/home/jeremy/Dev/Nek5000 CEED_DIR=/home/jeremy/Dev/libCEED \
	 FC=mpif77 CC=mpicc \
	 ./nek-examples.sh -e bp1
Box size not specified. Try setting it with -b option.
Try ./nek-examples.sh -h for more help.
make: *** [Makefile:42: nek] Error 1
```

`./nek-examples.sh -m` in `examples/nek5000` folder - succeeds

`make prove` at  top level - fails as described above without special variable

Reply:
@thilinarmtb What's your timeline to address the above so we can merge?

Reply:
@jedbrown : I will try to finish before this weekend. Bit busy these days.

Reply:
@jedbrown: It is not possible for genbox to put the boxes in a designated dir. NEKBOX_DIR is a default that needs to be set in makefile. How do you suggest we resolve this?

Reply:
If it were me, I'd add a `-o`/`--output` option to `genbox`. We shouldn't allow our productivity to suffer for eternity because Paul neglected a command-line utility best practice in 1994. If not that, the obvious thing would be to have a makefile rule that runs `genbox` in a temp directory and moves the result. We can't use `examples/nek5000/Makefile` because `makenek` writes its own (I still don't understand why that's considered acceptable), but we can write such a thing into the shell script or a higher level makefile (e.g., `examples/Makefile`). The complexity cannot be forced upon the user.

Reply:
Sure, I will fix this thing by tomorrow.